### PR TITLE
[posix] fix missing `nullptr` pointer check

### PR DESCRIPTION
**Pull request by cw8466**
Avoid segmentation fault for interfaces without link.

### CODE REVIEW

File: src/posix/platform/infra_if.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ bool otPlatInfraIfHasAddress(uint32_t aInfraIfIndex, const otIp6Address *aAddres
     {
         struct sockaddr_in6 *ip6Addr;
 
-        if (if_nametoindex(addr->ifa_name) != aInfraIfIndex || addr->ifa_addr->sa_family != AF_INET6)
+        if (if_nametoindex(addr->ifa_name) != aInfraIfIndex || addr->ifa_addr == NULL || addr->ifa_addr->sa_family != AF_INET6)
```
Comment (bukepo) [inline src/posix/platform/infra_if.cpp:76]:
```suggestion
        if (if_nametoindex(addr->ifa_name) != aInfraIfIndex || addr->ifa_addr == nullptr || addr->ifa_addr->sa_family != AF_INET6)
```

**Review by bukepo — COMMENTED** [review]
could you please fix the style issue?

**Review by wgtdkp — APPROVED** [review]
Thanks!
LGTM 👍

**Review by superwhd — APPROVED** [review]
LGTM 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**